exporter/containerimage/writer.go

```diff
@@ -235,6 +235,13 @@ func (ic *ImageWriter) commitDistributionManifest(ctx context.Context, ref cache
 	}
 
 	for i, desc := range remote.Descriptors {
+		// oci supports annotations but don't export internal annotations
+		if oci {
+			delete(desc.Annotations, "containerd.io/uncompressed")
+			delete(desc.Annotations, "buildkit/createdat")
+		} else {
+			desc.Annotations = nil
```

Comment:
@tonistiigi Cross-repo mount fails in the test. `dockerPusher` doesn't support it without Annotations. 
```
--- FAIL: TestIntegration/TestLazyImagePush/worker=containerd-1.3 (1.26s)
        client_test.go:2283: 
            	Error Trace:	client_test.go:2283
            	            				run.go:171
            	Error:      	Should be true
            	Test:       	TestIntegration/TestLazyImagePush/worker=containerd-1.3
            	Messages:   	unexpected error <nil>
```

https://github.com/containerd/containerd/blob/caad53dba7dbedbb5cdb95c54ec46c2d751a53e6/remotes/docker/pusher.go#L131

Configuring the exporters used in `TestLazyImagePush` to push OCI image (with `oci-mediatypes` Attr or something) makes the test happy?

Reply:
@ktock Thanks for debug. Do you want to make a follow-up that makes `oci-mediatypes=false` and stargz invalid combination as they don't seem to be compatible then if annotation is required.

Reply:
@tonistiigi I'm willing to contribute to this issue, but this patch doesn't seem directly related to stargz config (actually, TestLazyImagePush isn't a test for stargz images). Discarding the descriptor annotations seem to end up disabling cross-repo mounts of any types of lazyrefs (including non-stargz layers).

Roughly reading through codes around exporter/pusher, I currently think we need changes on push handlers in `util/push`, based on the following understanding (please tell me if I'm missing something): 

For non-lazy layers, even if the layer descriptors lost annotations (this occurs by this patch), `containerd.io/distribution.source...` are recovered from that layer's entry stored in the content store, during `(util/push).annotateDistributionSourceHandler`. But lazyrefs don't have these entries in the content store so this recovery doesn't work. So we might need additional logic something like recovering distribution annotations from the original manifest (not layers) blob.

cc: @sipsma

Reply:
@ktock You're right. We should make sure `containerd.io/distribution.source` do not end up in image manifests as well but if I remove them from here it would indeed disable cross-repo push as well. So we need some other way to pass annotations, not through the json.